src/main/java/org/checkerframework/checker/builder/TypesafeBuilderAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,127 @@
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.framework.util.MultiGraphQualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import javax.lang.model.element.AnnotationMirror;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The annotated type factory for the typesafe builder checker. Primarily responsible
+ * for the subtyping rules between @CalledMethod annotations.
+ */
+public class TypesafeBuilderAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+    /**
+     * Canonical copies of the top and bottom annotations. Package private to permit access
+     * from the Transfer class.
+     */
+    final AnnotationMirror TOP, BOTTOM;
+
+    /**
+     * Default constructor matching super. Should be called automatically.
+     */
+    public TypesafeBuilderAnnotatedTypeFactory(final BaseTypeChecker checker) {
+        super(checker);
+        TOP = createCalledMethods();
+        BOTTOM = AnnotationBuilder.fromClass(elements, CalledMethodsBottom.class);
+        this.postInit();
+    }
+
+    /** Creates a @CalledMethods annotation whose values are the given strings. */
+    public AnnotationMirror createCalledMethods(final String... val) {
+        AnnotationBuilder builder = new AnnotationBuilder(processingEnv, CalledMethods.class);
+        Arrays.sort(val);
+        builder.setValue("value", val);
+        return builder.build();
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(final MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+        return new TypesafeBuilderQualifierHierarchy(factory);
+    }
+
+    /**
+     * The qualifier hierarchy is responsible for lub, glb, and subtyping between qualifiers without
+     * declaratively defined subtyping relationships, like our @CalledMethods annotation.
+     */
+    private class TypesafeBuilderQualifierHierarchy extends MultiGraphQualifierHierarchy {
+        public TypesafeBuilderQualifierHierarchy(final MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+            super(factory);
+        }
+
+        @Override
+        public AnnotationMirror getTopAnnotation(final AnnotationMirror start) {
+            return TOP;
+        }
+
+        /**
+         * GLB in this type system is set union of the arguments of the two annotations,
+         * unless one of them is bottom, in which case the result is also bottom.
+         */
+        @Override
+        public AnnotationMirror greatestLowerBound(final AnnotationMirror a1, final AnnotationMirror a2) {
+            if (AnnotationUtils.areSame(a1, BOTTOM) || AnnotationUtils.areSame(a2, BOTTOM)) {
+                return BOTTOM;
+            }
+            Set<String> a1Val = getValueOfAnnotationWithStringArgument(a1).stream().collect(Collectors.toSet());
```

Comment:
Why use a stream here?  I'd prefer just `new LinkedHashSet<>(...)` unless there's a good reason to do otherwise.  Same for other instances of `xxx.stream().collect(Collectors.toSet())`